File: src/main/java/chinchulin/varano/Models/Course.java
package chinchulin.varano.Models;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "courses")
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_course;

    @Column(name = "name", nullable = false, unique = true, length = 30)
    private String name;

    @Column(name = "active", nullable = false)
    private Boolean active;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL)
    private List<Subject> subjects = new ArrayList<>();

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL)
    private List<AcademicRecord> records = new ArrayList<>();

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL)
    private List<Student> students = new ArrayList<>();

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL)
    private List<Module> modules = new ArrayList<>();
}

File: src/main/kotlin/com/project/varano/models/Course.kt
package com.project.varano.models

import jakarta.persistence.*
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

@Data
@Builder
@NoArgsConstructor
@Entity
@Table(name = "courses")
data class Course(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_course")
    private val id: Long,
    @Column(nullable = false, unique = true, length = 30)
    private val name: String,
    private val active: Boolean = true,

    @OneToMany(mappedBy = "course", cascade = [CascadeType.ALL])
    private val subjects: MutableSet<Subject> = mutableSetOf(),

    @OneToMany(mappedBy = "course", cascade = [CascadeType.ALL])
    private val records: MutableSet<AcademicRecord> = mutableSetOf(),

    @OneToMany(mappedBy = "course", cascade = [CascadeType.ALL])
    private val students: MutableSet<Student> = mutableSetOf(),

    @OneToMany(mappedBy = "course", cascade = [CascadeType.ALL])
    private val modules: MutableSet<Module> = mutableSetOf()
)
